xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authorization/xwiki-platform-security-authorization-bridge/src/main/java/org/xwiki/security/authorization/internal/DefaultSecurityEntryReader.java

```diff
@@ -253,17 +253,16 @@ private XWikiDocument getDocument(DocumentReference documentReference) throws Au
 
     /**
      * @param wikiReference the wiki to look for owner
-     * @return a reference to the owner of the wiki
-     * @throws AuthorizationException if the owner could not be retrieved.
+     * @return a reference to the owner of the wiki or {@code null}
      */
-    private DocumentReference getWikiOwner(WikiReference wikiReference) throws AuthorizationException
+    private DocumentReference getWikiOwner(WikiReference wikiReference)
     {
         XWikiContext context = this.xcontextProvider.get();
         String wikiOwner;
         try {
             wikiOwner = context.getWiki().getWikiOwner(wikiReference.getName(), context);
         } catch (XWikiException e) {
-            throw new AuthorizationException(wikiReference, "Could not retrieve the owner of this wiki", e);
+            wikiOwner = null;
```

Comment:
An XWikiException does not always mean that the wiki does not exist, would be more accurate to check if the wiki exist before calling getWikiOwner.

But the best would probably actually to instead use WikiDescriptorManager#getById (after making sure that getById return null and not an exception when the wiki does not exist).

Reply:
> WikiDescriptorManager#getById (after making sure that getById return null and not an exception when the wiki does not exist).

So it does indeed return null, now it could throw an exception if the document containing the XWikiServerClass cannot be loaded for some reason. Could it create problems if there's wrong rights on this document?

Reply:
> now it could throw an exception if the document containing the XWikiServerClass cannot be loaded for some reason

And as I said, it's OK, we do want to know when an important document cannot be loaded for an unexpected reason. The problem of your current proposal is that you are hiding those cases.

> Could it create problems if there's wrong rights on this document?

Not sure what you mean.

Reply:
> >Could it create problems if there's wrong rights on this document?
> Not sure what you mean.

Forget it, was doing a mistake

Reply:
> But the best would probably actually to instead use WikiDescriptorManager#getById (after making sure that getById return null and not an exception when the wiki does not exist).

So FTR as discussed on the chat this solution cannot be used because the WikiDescriptorCache is also invalidated with a listener and we cannot guarantee which listener is called first.